x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/NodeDecisions.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.autoscaling.storage;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+class NodeDecisions implements ToXContentObject, Writeable {
+
+    private final List<NodeDecision> canAllocateDecisions;
+    private final List<NodeDecision> canRemainDecisions;
+
+    NodeDecisions(List<NodeDecision> canAllocateDecisions, List<NodeDecision> canRemainDecisions) {
+        this.canAllocateDecisions = canAllocateDecisions;
+        this.canRemainDecisions = canRemainDecisions;
+    }
+
+    NodeDecisions(StreamInput in) throws IOException {
+        canAllocateDecisions = in.readList(NodeDecision::new);
+        canRemainDecisions = in.readList(NodeDecision::new);
+    }
+
+    List<NodeDecision> canAllocateDecisions() {
+        return canAllocateDecisions;
+    }
+
+    List<NodeDecision> canRemainDecisions() {
```

Comment:
Does this have to be a list? I would imagine that there is only a decision for one node, the current node of the shard. I did not try to check this though, perhaps you can look into that.

Reply:
Yes, from the tests I can see that it we can have multiple `can_remain` decisions.

Reply:
Can you explain how those multiple can remain decisions, i.e., for multiple nodes, comes about? Can you perhaps point me to how I can verify this easily using the test you mention here?

Reply:
@henningandersen I was wrong! The tests that produced multiple `can_remain` decisions were serialization tests with randomized input. There we no cases with multiple `can_remain` decisions in any integration tests. I've changed `NodeDecisions` to store a single instance of `canRemainDecision` and adjusted `ReactiveStorageDeciderService` to yield a single `canRemainDecision` or null instead of a singleton/empty list.